spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWriterV2.java

```diff
@@ -226,4 +229,100 @@ public void testWriteWithCaseSensitiveOption() throws NoSuchTableException, Pars
     fields = Spark3Util.loadIcebergTable(sparkSession, tableName).schema().asStruct().fields();
     Assert.assertEquals(4, fields.size());
   }
+
+  @Test
+  public void testMergeSchemaIgnoreDowncast() throws Exception {
+    //// test long to int ignore case
+    sql(
+        "ALTER TABLE %s SET TBLPROPERTIES ('%s'='true')",
+        tableName, TableProperties.SPARK_WRITE_ACCEPT_ANY_SCHEMA);
+
+    Dataset<Row> bigintDF =
+        jsonToDF(
+            "id bigint, data string",
+            "{ \"id\": 1, \"data\": \"a\" }",
+            "{ \"id\": 2, \"data\": \"b\" }");
+
+    bigintDF.writeTo(tableName).append();
+
+    assertEquals(
```

Comment:
I'm slightly worried this will be true regardless of the types. I think ... haven't tested this, that 1L and 1 will pass AssertEquals in our rows.

Probably we should also assert on schema

Reply:
Thanks for calling this out, I verified it indeed fails when asserting against rows with the int variant like `row(1, "a")` so I think it doing the right thing.

I'll assert on the schema as well 👍